### PR TITLE
CMP-3859: Add e2e test (60342) for disabled timeout with zero value

### PR DESCRIPTION
**Pull request by taimurhafeez**
test case to verify that setting the timeout to 0s in a ScanSetting disables the timeout functionality. This ensures scans can complete without timeout errors when timeout is explicitly set to zero.

The test creates a ScanSetting with a timeout set to 0 seconds and a maxRetryOnTimeout set to 2, binds it to the ocp4-cis profile, and verifies that all scans complete successfully in the DONE phase without any ERROR results due to timeouts.

To implement this PR:

 `make e2e-parallel E2E_GO_TEST_FLAGS="-v -run TestTimeoutDisabledWithZeroValue"`

Tested on OCP 4.22 and expected output:
```
=== RUN   TestTimeoutDisabledWithZeroValue
=== PAUSE TestTimeoutDisabledWithZeroValue
=== CONT  TestTimeoutDisabledWithZeroValue
2026/04/24 10:56:31 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:56:35 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:56:40 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:56:45 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:56:50 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:56:55 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:57:01 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: AGGREGATING
2026/04/24 10:57:06 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: AGGREGATING
2026/04/24 10:57:16 ComplianceScan ready (DONE)
2026/04/24 10:57:16 All scans in ComplianceSuite have finished (test-timeout-disabled-with-zero-value-binding)
--- PASS: TestTimeoutDisabledWithZeroValue (51.85s)
PASS

```

### CODE REVIEW

File: tests/e2e/framework/main_entry.go [hunk 1]
```diff
@@ -123,6 +123,7 @@ func (f *Framework) SetUp() error {
 	err = f.createInvalidMachineConfigPool("e2e-invalid")
 	if err != nil {
 		return fmt.Errorf("failed to create Machine Config Pool %s: %w", "e2e-invalid", err)
+		//log.Printf("did not find machineconfig e2e-invalid, but still proceeding")
```
Comment (xiaojiey) [inline tests/e2e/framework/main_entry.go]: Delete comment

File: tests/e2e/framework/common.go [hunk 2]
```diff
@@ -818,6 +818,7 @@ func (f *Framework) createInvalidMachineConfigPool(n string) error {
 		})
 	if createErr != nil {
 		return fmt.Errorf("failed to create Machine Config Pool %s: %w", n, createErr)
+		//log.Printf("did not find machineconfig e2e-invalid, but still proceeding")
```
[the remark this answers is not in the document]
Comment (taimurhafeez) [reply tests/e2e/framework/common.go]: done.

File: tests/e2e/framework/common.go [hunk 3]
```diff
@@ -831,11 +832,13 @@ func (f *Framework) cleanUpMachineConfigPool(n string) error {
 	err := f.Client.Get(context.TODO(), types.NamespacedName{Name: n}, p)
 	if err != nil {
 		return fmt.Errorf("failed to get Machine Config Pool %s for cleanup: %w", n, err)
+
```
[the remark this answers is not in the document]
Comment (taimurhafeez) [reply tests/e2e/framework/common.go]: removed.

**Review by rhmdnd — COMMENTED** [review]
Looks good to me, just need to remove the comment in main_entry.go

**Review by rhmdnd — APPROVED** [review]
/lgtm

**Comment by rhmdnd** [comment]
/retest-required

Looks like some of the scans timed out... We might want to consider breaking the test suite into smaller pieces based on functionality. That might help us isolate problematic tests.

**Comment by rhmdnd** [comment]
> > Another issue with a scan timing out. This time, waiting for the compliance suite to be created, which seems odd..
> > ```
> > 2025/12/03 00:41:05 waiting for availability of test-suspend-scan-setting-binding compliancesuite
> > 2025/12/03 00:41:05 waiting for availability of test-suspend-scan-setting-binding compliancesuite
> >     main_test.go:1770: compliancesuites.compliance.openshift.io "test-suspend-scan-setting-binding" not found
> > --- FAIL: TestSuspendScanSetting (1800.44s) 
> > ```
> 
> Strange. Is it from the CI run or your local run?

From CI - https://storage.googleapis.com/test-platform-results/pr-logs/pull/ComplianceAsCode_compliance-operator/982/pull-ci-ComplianceAsCode-compliance-operator-master-e2e-aws-serial/1995970267255410688/build-log.txt

**Review by rhmdnd — APPROVED** [review]
/lgtm

**Review by rhmdnd — APPROVED** [review]
Only minor comments about cosmetic issues that we can cleanup in a follow on PR.

**Comment by rhmdnd** [comment]
I wonder if this needs to be rebased to pass enterprise contract testing?

**Review by rhmdnd — DISMISSED** [review]

**Review by rhmdnd — COMMENTED** [review]
Looks like we may have a potential issue with the test:

```
2026/03/03 15:21:55 Waiting for availability of ocp4-cis compliancescan
2026/03/03 15:21:55 Waiting for availability of ocp4-cis compliancescan
2026/03/03 15:21:55 All scans in ComplianceSuite have finished (test-timeout-disabled-with-zero-value-binding)
=== NAME  TestTimeoutDisabledWithZeroValue
    main_test.go:5420: failed to get scan ocp4-cis: compliancescans.compliance.openshift.io "ocp4-cis" not found
--- FAIL: TestTimeoutDisabledWithZeroValue (1815.07s)
FAIL
FAIL	github.com/ComplianceAsCode/compliance-operator/tests/e2e/parallel	2162.543s
```

**Comment by rhmdnd** [comment]
Putting this back on my review queue until we can investigate the potential issue with the new test.

**Review by rhmdnd — APPROVED** [review]
Looks good to me. The only thing to note is the whitespace changes across `crd.yaml`, `deploy_rbac.yaml`, and `CheckType` in `main_test.go`. I like to keep whitespace separate from functional changes since it keeps the git history cleaner, and makes it easier to parse the history of a file.
